interface/ceed.c

```diff
@@ -594,8 +594,40 @@ int CeedSetData(Ceed ceed, void *data) {
 /// @addtogroup CeedUser
 /// @{
 
+/**
+  @brief Get the list of avaliable resource names for Ceed contexts
+  Note: The caller is responsible for freeing the resources and priorities arrays,
+          but should not free the contents of the resources array.
```

Comment:
We should explain how to free. `CeedFree` is in `ceed/backend.h` and I don't think we want to require that `CeedMalloc` be freeable with `free()`. So we should probably use raw `malloc` here.